### PR TITLE
feat(auth): Added support for creating tenant-scoped custom tokens

### PR DESCRIPTION
**Pull request by hiranya911**
* Implemented support for creating tenant-scoped custom tokens.
* Moved custom token creation APIs to the `AbstractFirebaseAuth` parent class so they are available on both `FirebaseAuth` and `TenantAwareFireabseAuth`.
* Cleaned up and refactored the unit tests to check the custom token creation functionality via both public interfaces. See the new `CustomTokenTest` where I use parameterized tests for this purpose.

### CODE REVIEW

File: FirebaseAdmin/FirebaseAdmin.Tests/Auth/CustomTokenTest.cs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            {
+                Credential = GoogleCredential.FromAccessToken("test-token"),
+                ProjectId = "project1",
+            };
+            var auth = config.CreateAuth(options);
+
+            var ex = await Assert.ThrowsAsync<InvalidOperationException>(
+                () => auth.CreateCustomTokenAsync("user1"));
+
+            var errorMessage = "Failed to determine service account ID. Make sure to initialize the SDK "
```
Comment (MathBunny) [inline FirebaseAdmin/FirebaseAdmin.Tests/Auth/CustomTokenTest.cs:84]: Just a thought, this exact string appears in three other places in the codebase already, maybe if it were a const string the code would be cleaner.

File: FirebaseAdmin/FirebaseAdmin/Auth/Multitenancy/TenantManager.cs [hunk 2]
```diff
@@ -336,8 +337,9 @@ void IDisposable.Dispose()
 
         internal static TenantManager Create(FirebaseApp app)
         {
-            var args = new Args
+            var args = new Args()
```
Comment (MathBunny) [inline FirebaseAdmin/FirebaseAdmin/Auth/Multitenancy/TenantManager.cs:340]: Just wondering why was this required?

Comment (hiranya911) [reply FirebaseAdmin/FirebaseAdmin/Auth/Multitenancy/TenantManager.cs:340]: It's just a cleaner and shorter syntax. We ought to use it whenever possible. Not a must though.

File: FirebaseAdmin/FirebaseAdmin.Tests/Auth/CustomTokenTest.cs (hunk 1 above)
Comment (hiranya911) [reply FirebaseAdmin/FirebaseAdmin.Tests/Auth/CustomTokenTest.cs:84]: I believe it's in `IAMSigner` and a couple of test cases. I'd prefer to have the error message repeated in the tests to make sure developers see the correct error message at runtime (it actually contains some useful information for debugging). But if it starts appearing in more places, we can look into introducing a constant of some sorts.
